crates/librqbit/src/watch.rs

```diff
@@ -109,23 +121,37 @@ fn watch_thread(
                 return Ok(());
             }
         }
-        for path in ev.paths {
-            if path.extension().and_then(|e| e.to_str()) != Some("torrent") {
-                trace!(?path, "ignoring path");
-                continue;
-            }
-            let add = match read_and_validate_torrent(&path) {
-                Ok(add) => add,
-                Err(e) => {
-                    warn!(?path, "error validating torrent: {e:#}");
-                    continue;
+
+        let _results = ev
+            .paths
+            .iter()
+            .filter_map(|path| match path.extension() {
+                Some(x) if x.to_str() == Some("torrent") => match read_and_validate_torrent(path) {
```

Comment:
it would probably be a bit shorter an more efficient if you do

```rust
match path.extension().and_then(|e| e.to_str())? {
   "torrent" => ...,
   "magnet" => ...,
}
```

Reply:
right, probably you don't want `?` as if by any reason a file without extension would land in the watch folder you would return and not process the rest of files right?

Changed it to `match path.extension().and_then(|e| e.to_str()) { ...}`

Reply:
As it's part of filter_map, the ? would only apply to the callback, not the outer function, so it should work

Reply:
Ok learned something new, thanks, i will change that

Reply:
Oh I somehow I forgot that it was inside a closure, that's why, it returns from there.

Done